src/System.IO.Pipelines.Networking.Tls/Internal/OpenSsl/LockStore.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.IO.Pipelines.Networking.Tls.Internal.OpenSsl
+{
+    internal unsafe static class LockStore
+    {
+        private static SemaphoreSlim[] _locks;
```

Comment:
Because nowhere could I get a definitive answer to the question "If thread A locks lockid 2 will thread A be the only thread that unlocks it?" As such we can't use a mutex unless I take the risk or go through all of OpenSsl or get a hard yes/no from someone. Unlocking a mutex from another thread is undefined... and not safe